### PR TITLE
Update cache protocol

### LINKED ISSUE #29: We have to clear cache if the protocol is changed between ssh and https.
Check Cache::fetch. If the repo was already cloned (it is in the cache) and is just being updated. We need to make sure when the protocol changes to apply that to the cache metadata by:

- Changing the local data
- Cloning again (fallback)

### PR DESCRIPTION
**Pull request by Gor027**
Information about the dependency protocol is stored either in the remote url of the local cache or in the lockfile generated by protofetch. As the lockfile cannot be updated in `cache::fetch`, user should manually force re-creation of the lockfile after modifing dependency protocol.

It should fix #29.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,4 +15,7 @@ target/
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
 
+# Generated by IDEs
+.idea
```
Comment (rtimush) [inline .gitignore:19]: Let's not add IDE-specific patterns to the repository, they fit much better in `$XDG_CONFIG_HOME/git/ignore` or wherever the global git config `core.excludesfile` points to.

File: src/api/mod.rs [hunk 2]
```diff
@@ -32,9 +32,9 @@ impl Protofetch {
     }
 
     /// Fetches dependencies defined in the toml configuration file
-    pub fn fetch(&self, ignore_lock_file: bool) -> Result<(), Box<dyn Error>> {
+    pub fn fetch(&self, force_lock: bool) -> Result<(), Box<dyn Error>> {
         do_fetch(
-            ignore_lock_file,
+            force_lock,
```
Comment (rtimush) [inline src/api/mod.rs:37]: It's nitpicking, but I'm not sure I agree with this rename. `force_lock` can be read as "force lock file usage" which is not what it does. Although, maybe `ignore_lock_file` is not the best name either, as it does not reflect the fact that the lock file is updated. Maybe something along the lines of `force_lock_update`...

Comment (Gor027) [reply src/api/mod.rs:37]: This seems to be resolved already in the master, so I will drop the commit.

File: src/cache.rs [hunk 3]
```diff
@@ -104,8 +104,23 @@ impl ProtofetchGitCache {
...
+            if url != new_url {
+                // If true then the protocol was updated before updating the cache.
+                trace!(
+                    "Updating remote existing url {} to new url {}",
+                    url,
+                    new_url
+                );
+                repo.remote_set_url("origin", &new_url)?;
+            }
+        }
```
Comment (Gor027) [inline src/cache.rs:122]: Now, after some changes and the removal of `cache::fetch` function, I am not sure, where is the best place to apply the protocol change update. @rtimush as you are more familiar with the code, what is your opinion about it?

Comment (rtimush) [reply src/cache.rs:122]: I suppose https://github.com/coralogix/protofetch/blob/142d46a09fbcb2ef2ea1c2384830dc908158991f/src/git/cache.rs#L93-L95 is now the right place.

Comment (Gor027) [reply src/cache.rs:122]: Thanks.

File: README.md [hunk 4]
```diff
@@ -53,6 +53,10 @@ Protofetch is also released to [crates.io](https://crates.io/crates/protofetch),
 ```sh
 # Fetch proto sources, updating the lock file if needed.
 protofetch fetch
+
+# -f forces lock file to be generated in every run
+# Should be used in cases when information like protocol is updated to have up-to-date lockfile.
+protofetch fetch -f
```
Comment (rtimush) [inline README.md:59]: Let's remove this note, `-f` is now deprecated, and it is also not necessary anymore to trigger a lockfile update.

**Review by rtimush — APPROVED** [review]

**Pull request merged**